internal/dnfjson/dnfjson.go

```diff
@@ -0,0 +1,200 @@
+		Arch:             arch,
+		CacheDir:         cacheDir,
+	}
+}
+
+// Depsolve the given packages with explicit excludes using the solver configuration
+func (s *Solver) Depsolve(includes []string, excludes []string) (*Result, error) {
+	req := Request{
+		Command: "depsolve",
+		Arguments: Arguments{
+			PackageSpecs: includes,
+			ExcludSpecs:  excludes,
+			Solver:       *s,
+		},
+	}
+	return run(req)
+}
+
+func (s *Solver) FetchMetadata() (*Result, error) {
+	req := Request{
+		Command: "dump",
+		Arguments: Arguments{
+			Solver: *s,
+		},
+	}
+	return run(req)
+}
+
+// Repository configuration for resolving dependencies for a set of packages. A
+// Solver needs at least one RPM repository configured to be able to depsolve.
+type RepoConfig struct {
+	ID             string `json:"id"`
+	Name           string `json:"name,omitempty"`
+	BaseURL        string `json:"baseurl,omitempty"`
+	Metalink       string `json:"metalink,omitempty"`
+	MirrorList     string `json:"mirrorlist,omitempty"`
+	GPGKey         string `json:"gpgkey,omitempty"`
+	IgnoreSSL      bool   `json:"ignoressl"`
+	SSLCACert      string `json:"sslcacert,omitempty"`
+	SSLClientKey   string `json:"sslclientkey,omitempty"`
+	SSLClientCert  string `json:"sslclientcert,omitempty"`
+	MetadataExpire string `json:"metadata_expire,omitempty"`
+}
+
+// Request command and arguments for dnf-json
+type Request struct {
+	Command   string    `json:"command"`
+	Arguments Arguments `json:"arguments"`
+}
+
+// Arguments for a dnf-json request
+type Arguments struct {
+	// Packages to depsolve
+	PackageSpecs []string `json:"package-specs"`
+
+	// Packages to exclude from results
+	ExcludSpecs []string `json:"exclude-specs"`
+
+	// Solver configuration
+	Solver
+}
+
+// Result of a dnf-json depsolve run
+type Result struct {
+	// Repository checksums
+	Checksums map[string]string `json:"checksums"`
+
+	// Resolved package dependencies
+	Dependencies []PackageSpec `json:"dependencies"`
+}
+
+// Package specification
+type PackageSpec struct {
+	Name           string `json:"name"`
+	Epoch          uint   `json:"epoch"`
+	Version        string `json:"version,omitempty"`
+	Release        string `json:"release,omitempty"`
+	Arch           string `json:"arch,omitempty"`
+	RepoID         string `json:"repo_id,omitempty"`
+	Path           string `json:"path,omitempty"`
+	RemoteLocation string `json:"remote_location,omitempty"`
+	Checksum       string `json:"checksum,omitempty"`
+	Secrets        string `json:"secrets,omitempty"`
+}
+
+// dnf-json error structure
+type Error struct {
+	Kind   string `json:"kind"`
+	Reason string `json:"reason"`
+}
+
+func (err Error) Error() string {
+	return fmt.Sprintf("DNF error occurred: %s: %s", err.Kind, err.Reason)
+}
+
+// Depsolve the given packages with explicit excludes using the given configuration
+func Depsolve(packages []string, excludes []string, repos []RepoConfig, modulePlatformID string, arch string, cacheDir string) (*Result, error) {
```

Comment:
Is this not redundant?

Reply:
It's a convenience.  Rather common in Go, especially in the standard library, and I've gotten used to expecting it personally.